pkg/cluster/delete.go

```diff
@@ -377,6 +378,58 @@ func (m *manager) deleteClusterMsiCertificate(ctx context.Context) error {
 	return err
 }
 
+func (m *manager) deleteIdentityFederations(ctx context.Context) error {
+	if m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile == nil {
+		return nil
+	}
+	platformWIRolesByRoleName := m.platformWorkloadIdentityRolesByVersion.GetPlatformWorkloadIdentityRolesByRoleName()
+	platformWorkloadIdentities := m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities
+
+	for _, identity := range platformWorkloadIdentities {
+		_, exists := platformWIRolesByRoleName[identity.OperatorName]
+		if !exists {
+			continue
+		}
+
+		identityResourceId, err := azure.ParseResourceID(identity.ResourceID)
+		if err != nil {
+			return err
+		}
+
+		federatedIdentityCredentialResourceName, err := m.getPlatformWorkloadIdentityFederatedCredName(identity)
+		if err != nil {
+			return err
+		}
+
+		_, err = m.clusterMsiFederatedIdentityCredentials.Get(
```

Comment:
I tried testing the PR locally, but I get a nil pointer dereference on this line because the fed creds client isn't initialized. I think we'll need to:

1. Make sure the cluster MSI clients are initialized before trying to use them during deletion
2. Additionally, account for the case where the cluster MSI is missing and we're therefore unable to delete the fed creds. We probably don't want to block cluster deletion on this but rather log a warning message saying we did what we could (i.e.: `Unable to delete platform workload identities' federated credentials because cluster MSI has already been deleted; continuing with rest of cluster deletion`) and then move forward.

@cadenmarchese, what do you think about #2? Does what I suggested sound good?

Reply:
@kimorris27 - for point 1, should we error out if the Fed Cred client is `nil`? Also, is this required, as we already error out during `initializeClusterMsiClients` as [here](https://github.com/Azure/ARO-RP/blob/84816de0e43d611efdfbfa1aef1bf7f40f4bedf8/pkg/cluster/clustermsi.go#L121) if the client is not created.

Reply:
That function currently isn't getting called at all during deletion, but to your point, I think we will need to initialize the fed creds client differently during deletion, i.e. create it if possible and log a warning and continue with cluster deletion if not possible.

When an RP backend goroutine picks up a cluster whose state has changed to `Deleting`, it goes straight to the `Delete` function without following any of the other "steps", as you can see [here](https://github.com/Azure/ARO-RP/blob/84816de0e43d611efdfbfa1aef1bf7f40f4bedf8/pkg/backend/openshiftcluster.go#L186). So I think the logic for the fed creds client initialization that I described in the previous paragraph might need to be incorporated here.

Reply:
@kimorris27 - Was there any reason we didn't initialize the Fed Cred client when creating a [New cluster manager](https://github.com/Azure/ARO-RP/blob/84816de0e43d611efdfbfa1aef1bf7f40f4bedf8/pkg/cluster/cluster.go#L316)? Just wanted to understand this before making the changes?

Reply:
@kimorris27 as long as the customer can clean up resources on their own, I don't think we need to block deletion on failure to clean up federated credentials, so your suggestion sounds good to me.

Reply:
Addressed both point 1 & 2.